File: leetcode/1-two-sum.cxx
// https://leetcode.com/problems/two-sum

class Solution {
public:
  vector<int> twoSum(vector<int> &xs, int t) {
    unordered_map<int, int> memo;
    for (int i = 0; i < xs.size(); i += 1) {
      int x = xs[i];
      auto yi = memo.find(t - x);
      if (yi != memo.end()) {
        return {yi->second, i};
      }
      memo[x] = i;
    }
  }
};

File: leetcode/1-two-sum.js
// https://leetcode.com/problems/two-sum

/**
 * @param {number[]} xs
 * @param {number} t
 * @return {number[]}
 */
const twoSum = (xs, t) => {
  const memo = new Map()
  for (const [x, i] of xs.map((x, i) => [x, i])) {
    const yi = memo.get(t - x)
    if (yi !== undefined) {
      return [yi, i]
    }
    memo.set(x, i)
  }
}
